xarray/conventions.py

```diff
@@ -438,15 +438,31 @@ def stackable(dim: Hashable) -> bool:
             )
         except Exception as e:
             raise type(e)(f"Failed to decode variable {k!r}: {e}")
-        if decode_coords in [True, "coordinates", "all"]:
+        if decode_coords in [True, "coordinates", "all", "coordinates_strict"]:
             var_attrs = new_vars[k].attrs
             if "coordinates" in var_attrs:
-                coord_str = var_attrs["coordinates"]
-                var_coord_names = coord_str.split()
-                if all(k in variables for k in var_coord_names):
-                    new_vars[k].encoding["coordinates"] = coord_str
-                    del var_attrs["coordinates"]
-                    coord_names.update(var_coord_names)
+                var_coord_names = var_attrs["coordinates"].split()
+                existing_coord_names = [c for c in var_coord_names if c in variables]
+                missing_coord_names = [c for c in var_coord_names if c not in variables]
+                if missing_coord_names:
```

Comment:
```suggestion
                existing_coord_names = [c for c in var_coord_names if c in variables]
                if not existing_coord_names:
                    missing_coord_names = [c for c in var_coord_names if c not in variables]
```
`missing_coord_names` looks like a error state calculation, we don't want to run that calculation if everything works as intended.

I wonder if these comparison can be done using `set`-comparison?

Reply:
> I wonder if these comparison can be done using `set`-comparison?

Yes, that greatly simplified the workflow.